packages/solid/src/render/component.ts

```diff
@@ -179,6 +179,7 @@ function resolveSource(s: any) {
 }
 
 export function mergeProps<T extends unknown[]>(...sources: T): MergeProps<T> {
+  sources = sources.map(source => typeof source === "function" ? createMemo(source) : source) as T
```

Comment:
alternatively, we can merge into a single loop, 
which will detect and transform in one go(full scan). 
```js
export function mergeProps(...sources) {
  let proxy = 0;
  for (let i = 0; i < sources.length; i++) {
    const s = sources[i];
    proxy ||= s && $PROXY in s;
    sources[i] = s === 'function' ? ((proxy = 1), createMemo(s)) : s;
  }

  if (proxy) {
    return new Proxy(
      {
        get(property) {
          for (let i = sources.length - 1; i >= 0; i--) {
            let v = resolveSource(sources[i])[property];
            if (v !== void 0) return v;
          }
        },
```

Reply:
Yeah probably this. Some needs iterate everything on a miss anyway.

Reply:
This PR was merged, and I was about to make another one with the optimized loop as in the comments above.

and I see that @ryansolid  took care of that   👍 
https://github.com/solidjs/solid/commit/a42a5f6e7150a981a979f84c2745a4a4ba03f6ec

Reply:
we should look into lazy memo for this case